### PR TITLE
Add support for onFlagClick

### PR DESCRIPTION
**Pull request by tomegz**

### CODE REVIEW

File: src/components/IntlTelInputApp.js [hunk 1]
```diff
@@ -860,6 +861,7 @@ class IntlTelInputApp extends Component {
       // need to hide dropdown when click on opened flag button
       this.toggleDropdown(false);
     }
+    this.props.onFlagClick();
```
Comment (patw0929) [inline src/components/IntlTelInputApp.js]: Should we check `onFlagClick` type before invoking it?

like this:
```js
if (typeof this.props.onFlagClick === 'function') {
  this.props.onFlagClick();
}
```

Comment (tomegz) [reply src/components/IntlTelInputApp.js]: I'd say it's up to you. I think that if users passes something that isn't a function it should break, so he knows what he did wrong. What are your thoughts?

Comment (tomegz) [reply src/components/IntlTelInputApp.js]: What you guys think about the naming convention inconsistency now? I mean on**Select**Flag vs onFlag**Click**. Should I stick to original convention and change it to on**Click**Flag?

Comment (nutboltu) [reply src/components/IntlTelInputApp.js]: @tomegz 
It looks like default value of this function is `() => {}`. It reduces the chance to breaking the component. I would suggest to change the default props value to `null`.

I'm fine with the naming `onFlagClick` But shouldn't we pass the event or any values as parameter? TBH I didn't find it's useful without passing specific values.

Comment (nutboltu) [reply src/components/IntlTelInputApp.js]: @tomegz 
I was thinking to pass the `event` as a parameter, So the parent component can have control with the target element using `event.target`.
#251 seems like we can also proceed without passing any parameters.

Comment (tomegz) [reply src/components/IntlTelInputApp.js]: @patw0929 @nutboltu Updated with passing event as argument

Comment (tomegz) [reply src/components/IntlTelInputApp.js]: Does it need any more work before merging?

**Pull request merged**